Makefile

```diff
@@ -8,24 +8,25 @@
 
 .PHONY: test test-unit test-e2e build goreleaser release clean
 os ?= $(shell uname -s | tr '[:upper:]' '[:lower:]')
+arch ?= $(shell go env GOARCH | tr '[:upper:]' '[:lower:]')
 
 test: test-unit test-e2e test-integration
 
 test-unit:
 	go test -v ./...
 
-test-e2e: dist/kubectl-neat_$(os) 
+test-e2e: dist/kubectl-neat_$(os)_$(arch)
 	bats ./test/e2e-cli.bats
 
-test-integration: dist/kubectl-neat_$(os).tar.gz dist/kubectl-neat_$(os)*/kubectl-neat dist/checksums.txt
+test-integration: dist/kubectl-neat_$(os)_$(arch).tar.gz dist/kubectl-neat_$(os)_$(arch)*/kubectl-neat dist/checksums.txt
 	bats ./test/e2e-kubectl.bats
 	bats ./test/e2e-krew.bats
 
-build: dist/kubectl-neat_$(os)
+build: dist/kubectl-neat_$(os)_$(arch)
 
 SRC = $(shell find . -type f -name '*.go' -not -path "./vendor/*")
 dist/kubectl-neat_%: $(SRC)
-	GOOS=$* go build -o dist/$(@F)
+	GOOS=$(os) GOARCH=$(arch) go build -o dist/$(@F)
```

Comment:
In order to support cross-compilation, this target should take into account the *requested* os/arch, while it currently uses the *runtime* os/arch.

Reply:
@itaysk I added a function to split the `$*` argument into _requested_ os/arch for the build flags